modelserving/dev/tasks/build-images

```diff
@@ -17,5 +17,92 @@ if [[ -z "${TAG:-}" ]]; then
+fi
+
+dev/tasks/download-model
+
+# Note we do not push or load the "base" llama-server images (we do not pass BUILDX_ARGS)
+# This is because these are intermediate images
+if build_for_architecture cpu; then
+  docker buildx build \
+    -f images/llamacpp-server/Dockerfile \
+    --target llamacpp-server \
+    -t llamacpp-server-cpu:${TAG} \
+    --build-arg BASE_IMAGE=debian:latest \
+    --build-arg BUILDER_IMAGE=debian:latest \
+    --build-arg "CMAKE_ARGS=-DGGML_RPC=ON" \
+    --progress=plain .
+fi
+
+if build_for_architecture cuda; then
+  # -DCMAKE_EXE_LINKER_FLAGS=-Wl,--allow-shlib-undefined allows us to build in a container without all the CUDA libraries present
+  # These flags mirror the flags in the llama.cpp github-action: https://github.com/ggml-org/llama.cpp/blob/master/.github/workflows/build.yml
+
+  docker buildx build \
+    -f images/llamacpp-server/Dockerfile \
+    --target llamacpp-server \
+    -t llamacpp-server-cuda:${TAG} \
+    --build-arg BASE_IMAGE=nvidia/cuda:12.6.2-runtime-ubuntu24.04 \
+    --build-arg BUILDER_IMAGE=nvidia/cuda:12.6.2-devel-ubuntu24.04 \
+    --build-arg "CMAKE_ARGS=-DGGML_RPC=ON -DGGML_CUDA=ON -DCMAKE_CUDA_ARCHITECTURES=all -DCMAKE_EXE_LINKER_FLAGS=-Wl,--allow-shlib-undefined" \
+    --progress=plain .
+fi
+
+if build_for_architecture cpu; then
+  docker buildx build ${BUILDX_ARGS} \
+    -f images/llamacpp-gemma3-12b-it/Dockerfile \
+    -t ${IMAGE_PREFIX}llamacpp-gemma3-12b-it-cpu:${TAG} \
+    --build-arg BASE_IMAGE=llamacpp-server-cpu:${TAG} \
+    --progress=plain .
+fi
+
+if build_for_architecture cuda; then
+  docker buildx build ${BUILDX_ARGS} \
+    -f images/llamacpp-gemma3-12b-it/Dockerfile \
+    -t ${IMAGE_PREFIX}llamacpp-gemma3-12b-it-cuda:${TAG} \
+    --build-arg BASE_IMAGE=llamacpp-server-cuda:${TAG} \
+    --progress=plain .
+fi
+
+if build_for_architecture cpu; then
+  docker buildx build ${BUILDX_ARGS} \
+    -f images/llamacpp-server/Dockerfile \
+    --target rpc-server \
+    -t ${IMAGE_PREFIX}rpc-server-cpu:${TAG} \
+    --build-arg BASE_IMAGE=debian:latest \
+    --build-arg BUILDER_IMAGE=debian:latest \
+    --build-arg "CMAKE_ARGS=-DGGML_RPC=ON" \
+    --progress=plain .
+fi
+
+if build_for_architecture cuda; then
+  # -DCMAKE_EXE_LINKER_FLAGS=-Wl,--allow-shlib-undefined allows us to build in a container without all the CUDA libraries present
+  # These flags mirror the flags in the llama.cpp github-action: https://github.com/ggml-org/llama.cpp/blob/master/.github/workflows/build.yml
+
+  docker buildx build ${BUILDX_ARGS} \
+    -f images/llamacpp-server/Dockerfile \
+    --target rpc-server \
+    -t ${IMAGE_PREFIX}rpc-server-cuda:${TAG} \
+    --build-arg BASE_IMAGE=nvidia/cuda:12.6.2-runtime-ubuntu24.04 \
+    --build-arg BUILDER_IMAGE=nvidia/cuda:12.6.2-devel-ubuntu24.04 \
+    --build-arg "CMAKE_ARGS=-DGGML_RPC=ON -DGGML_CUDA=ON -DCMAKE_CUDA_ARCHITECTURES=all -DCMAKE_EXE_LINKER_FLAGS=-Wl,--allow-shlib-undefined" \
+    --progress=plain .
+fi
```

Comment:
I am surprised that the `llama.cpp` project doesn't have these images available somewhere we can copy from in the `deploy` step.

Reply:
They have some images: https://github.com/ggml-org/llama.cpp/blob/master/docs/docker.md

But I want to add the model.

That said, I probably should have at least looked at how they are building their images, might have saved me some time 😂